modis_tools/granule_handler.py

```diff
@@ -131,7 +130,7 @@ def download_from_urls(
         :returns Path to the newly downloaded file
         :rtype List[Path]
         """
-        urls = cls._coerce_to_list(one_or_many_urls, HttpUrl)
+        urls = cls._coerce_to_list(one_or_many_urls, AnyUrl)
```

Comment:
This is the actual bug fix. The rest is pre-commit. AnyUrl is the parent of AnyHttpUrl and the grandparent of HttpUrl, thus `isinstance` will cover all three causes with just one type